### PR TITLE
OPEN-2: Remove custom events in favor of Eloquent events

### PR DESCRIPTION
**Pull request by MerlijnCuypers**
Development for issue https://github.com/StadGent/laravel_site_opening-hours/issues/84
All events removed and replaced by default Eloquent model models triggered in observer classes.
Logic is placed in according services, as these events are meant to sync the data to VESTA and LOD, these are bundled under methode makeSyncJobsForExternalServices

### CODE REVIEW

File: app/Formatters/Openinghours.php [hunk 1]
```diff
@@ -104,9 +104,9 @@ protected function toHTML()
                     foreach ($schedule as $entry) {
                         $formattedSchedule .= "<div>$entry</div>";
                     }
-                } else {
-                    $formattedSchedule .= "<div>$schedule</div>";
-                }
+                    continue;
+                } 
+                $formattedSchedule .= "<div>$schedule</div>";
```
Comment (zero2one) [inline app/Formatters/Openinghours.php]: Can we use a template file to render the html version?

File: app/Http/Controllers/CalendarsController.php [hunk 2]
```diff
@@ -89,24 +87,24 @@ public function edit($id)
...
      * @param  int                       $id
      * @return \Illuminate\Http\Response
      */
     public function update(UpdateCalendarRequest $request, $id)
     {
         $input = $request->input();
 
-        $success = $this->calendars->update($id, $input);
-
         // If events are passed, bulk upsert them
```
Comment (zero2one) [inline app/Http/Controllers/CalendarsController.php]: "insert" instead of "upsert"?

File: app/Http/Controllers/CalendarsController.php [hunk 3]
```diff
@@ -149,8 +147,6 @@ public function destroy(DeleteCalendarRequest $request, $calendarId)
             return response()->json(['message' => 'De kalender werd niet verwijderd, er is iets foutgegaan.'], 400);
```
Comment (zero2one) [inline app/Http/Controllers/CalendarsController.php]: Make message translatable

File: app/Http/Controllers/ChannelController.php [hunk 4]
```diff
@@ -127,8 +127,6 @@ public function destroy($id)
             return response()->json(['message' => 'Het kanaal werd niet gevonden.'], 400);
```
Comment (zero2one) [inline app/Http/Controllers/ChannelController.php]: Make message translatable

File: app/Http/Controllers/ChannelController.php [hunk 5]
```diff
@@ -127,8 +127,6 @@ public function destroy($id)
...
         }
 
-        event(new ChannelDeleted($channel));
-
         $this->channels->delete($id);
 
         return response()->json(['Het kanaal werd verwijderd.']);
```
Comment (zero2one) [inline app/Http/Controllers/ChannelController.php]: Make message translatable

File: app/Http/Controllers/OpeninghoursController.php [hunk 6]
```diff
@@ -142,8 +136,6 @@ public function destroy(DeleteOpeninghoursRequest $request, $id)
         $success = $this->openinghours->delete($id);
 
         if ($success) {
-            event(new OpeninghoursDeleted($openinghours, $this->openinghours->isOpeninghoursRelevantNow($openinghours)));
-
             return response()->json(['message' => 'De openingsuren werden verwijderd']);
```
Comment (zero2one) [inline app/Http/Controllers/OpeninghoursController.php]: Make message translatable

File: database/factories/ModelFactory.php [hunk 7]
```diff
@@ -9,34 +9,34 @@
...
 
 /** @var \Illuminate\Database\Eloquent\Factory $factory */
 $factory->define(App\Models\SUser::class, function (Faker\Generator $faker) {
     static $password;
 
     return [
-        'name' => $faker->name,
-        'email' => $faker->unique()->safeEmail,
-        'password' => $password ?: $password = bcrypt('secret'),
+        'name'           => $faker->name,
```
Comment (zero2one) [inline database/factories/ModelFactory.php]: Bad practice to allign like this; if a new item is added and it has a longer key, all other lines need an update (= diff state in PR) to.

File: tests/controllers/QueryControllerTest.php [hunk 8]
```diff
@@ -75,7 +74,10 @@ public function testValidateWrongServiceUriParameterIsAFail()
         // undo service setter
         // will be restored for next test by setup
         $this->service = null;
-        $call          = $this->doRequest('GET', ['q' => 'now', 'serviceUri' => 'thisIsNotAServiceUri', 'date' => date('d-m-Y')]);
+        $call          = $this->doRequest(
```
Comment (zero2one) [inline tests/controllers/QueryControllerTest.php]: Only 1 space between variable and =

File: app/Formatters/Openinghours.php (hunk 1 above)
Comment (Jelle-S) [reply app/Formatters/Openinghours.php]: probably, but HTML output will be refactored in a different issue.

File: app/Http/Controllers/CalendarsController.php (hunk 2 above)
Comment (Jelle-S) [reply app/Http/Controllers/CalendarsController.php]: Comment comes from legacy code, but we might as well change it.

File: app/Http/Controllers/CalendarsController.php (hunk 3 above)
Comment (Jelle-S) [reply app/Http/Controllers/CalendarsController.php]: Again, comes from legacy code. Making text translatable will be handled in a different issue.

File: app/Observers/CalendarObserver.php [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+     * trigger jobs of Openinghours
+     * that will sync new values to external services
+     *
+     * @param  Openinghours $openinghours
+     * @return void
+     */
+    public function saved(Calendar $calendar)
+    {
+        app('OpeninghoursService')->makeSyncJobsForExternalServices($calendar->openinghours, 'update');
+    }
```
Comment (Jelle-S) [inline app/Observers/CalendarObserver.php]: Add deleted event

File: app/Observers/ChannelObserver.php [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+     * trigger jobs of Channel
+     * that will sync and remove values from external services
+     *
+     * @param  Channel $channel
+     * @return void
+     */
+    public function deleting(Channel $channel)
+    {
+        app('ChannelService')->makeSyncJobsForExternalServices($channel);
+    }
```
Comment (Jelle-S) [inline app/Observers/ChannelObserver.php]: Add saved event.

File: app/Providers/EventServiceProvider.php [hunk 11]
```diff
@@ -13,18 +13,9 @@ class EventServiceProvider extends ServiceProvider
      * @var array
      */
     protected $listen = [
-        'App\Events\OpeninghoursCreated' => [
-            'App\Listeners\HandleNewOpeninghours',
-        ],
-        'App\Events\OpeninghoursUpdated' => [
-            'App\Listeners\HandleUpdatedOpeninghours',
-        ],
         'App\Events\CalendarUpdated' => [
```
Comment (Jelle-S) [inline app/Providers/EventServiceProvider.php]: Delete these listeners

File: app/Http/Controllers/ChannelController.php (hunk 4 above)
Comment (MerlijnCuypers) [reply app/Http/Controllers/ChannelController.php]: will be done in issue  https://github.com/StadGent/laravel_site_opening-hours/issues/85

File: app/Http/Controllers/ChannelController.php (hunk 5 above)
Comment (MerlijnCuypers) [reply app/Http/Controllers/ChannelController.php]: will be done in issue  https://github.com/StadGent/laravel_site_opening-hours/issues/85

File: app/Http/Controllers/OpeninghoursController.php (hunk 6 above)
Comment (MerlijnCuypers) [reply app/Http/Controllers/OpeninghoursController.php]: will be done in issue  https://github.com/StadGent/laravel_site_opening-hours/issues/85

File: app/Http/Controllers/CalendarsController.php (hunk 3 above)
Comment (MerlijnCuypers) [reply app/Http/Controllers/CalendarsController.php]: will be done in issue  https://github.com/StadGent/laravel_site_opening-hours/issues/85

File: app/Observers/ChannelObserver.php (hunk 10 above)
Comment (MerlijnCuypers) [reply app/Observers/ChannelObserver.php]: will be reviewed in the refactoring of external services https://github.com/StadGent/laravel_site_opening-hours/issues/93 https://github.com/StadGent/laravel_site_opening-hours/issues/92

File: app/Observers/CalendarObserver.php (hunk 9 above)
Comment (MerlijnCuypers) [reply app/Observers/CalendarObserver.php]: will be reviewed in the refactoring of external services https://github.com/StadGent/laravel_site_opening-hours/issues/93 https://github.com/StadGent/laravel_site_opening-hours/issues/92

**Pull request merged**